### ISSUE TITLE
Cannot use babel-loader with plugins

### ISSUE THREAD

**Issue by steveetm** [webpack-contrib/grunt-webpack#138]
### Webpack/Grunt Configuration 

webpack.config.js
```js
{
    frontend: {
        resolve: {
            modules: [process.cwd(), 'client/', 'client/app/', 'node_modules']
        },
        entry: {
            'client': './client/app/app.js',
        },
        output: {
            path: path.resolve(process.cwd(), 'build/www'),
            filename: "client-[name].entry.js"
        },
        module: {
            rules: [
                {
                    test: /\.js/,
                    exclude: /node_modules/,
                    use: [
                        {
                            loader: 'babel-loader',
                            options: {
                                presets: ['babel-preset-es2015'],
                                plugins: ["transform-remove-strict-mode"],
                                ignore: ['client/vendor/sdk/ext']

                            },
                        },
                        {
                            loader: 'extjs-loader',
                            options: type
                        },
                    ]
                }
            ],


        }

    }
};
```
grunt-webpack config:
```json
 "webpack-dev-server": {
                    options: {
                        webpack: webpackConfig.frontend,
                        publicPath: '/',

                        hot: true,
                        compress: true,
                        port: 9000,
                        contentBase: path.resolve(process.cwd(), 'build/www/'),
                    },
                    start: {
                        webpack: {
                            devtool: "cheap-eval-sourcemap"
                        }
                    }

            }
```

### Expected Behavior
It should work

### Current Behavior
Running "webpack-dev-server" task
Warning: Object prototype may only be an Object or null: transform-remove-strict-mode Use --force to continue.

Aborted due to warnings.

### Possible Solution
Previous version of grunt-webpack is not messed with plugins options of babel config, while the latest one calls fixPlugin on babel's plugin config. Is it really neccesary to call fixPlugin on _every_ plugin config key?

### Your Environment

| software         | version
| ---------------- | -------
| webpack          |  2.3.3-2.5.1
| grunt-webpack    | 3.0.0
| grunt            | 1.0.1
| node             | 6.10.1
| npm              | 3.10.10
| Operating System |  OSX 10.12.4

**Comment by jdreesen** [comment]
It doesn't work for me either.

**Comment by danez** [comment]
the config in the first post is not correct it should be:

```js
 "webpack-dev-server": {
                    options: {
                        publicPath: '/',

                        hot: true,
                        compress: true,
                        port: 9000,
                        contentBase: path.resolve(process.cwd(), 'build/www/'),
                    },
                    webpack: Object.assign({
                        devtool: "cheap-eval-sourcemap"
                    }, webpackConfig.frontend)

            }
```

**Comment by brandonchinn178** [comment]
:+1:

Pretty sure the reason is starting in de1afbd3749048110da154e69df7c0cd8596a23c, grunt-webpack cloned plugin options using `Object.create` and copying over keys. The problem is:
```
x = [1,2,3]
instance = Object.create(x)
instance[0] = x[0]
console.log(instance) // Array { '0': 1 }
```
`instance` is no longer an Array, but an Array-like object. This fails in `babel-core` because it's expecting an [actual Array](https://github.com/babel/babel/blob/6.x/packages/babel-core/src/transformation/file/options/option-manager.js#L115)

This took a ton of digging and a lot of obscure error messages. Hope this can get fixed soon! I'll try to post up a workaround if I find one.

**Comment by mareksuscak** [comment]
What can I do in order to help you push this forward ASAP? It's the last missing piece of the puzzle for us.

**Comment by danez** [comment]
Will be fixed in 3.0.1 in some minutes. Sorry for the long delay I was on vacation.

**Comment by brandonchinn178** [comment]
@mareksuscak Sorry, I just saw all of your comments. I haven't been able to check GitHub a ton since work's gotten busier.
